### PR TITLE
Video stream

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

File: app/src/main/java/org/dasfoo/rover/android/client/video/MediaCodecHandler.java [hunk 1]
```diff
@@ -5,37 +5,60 @@
 import android.support.annotation.NonNull;
 import android.util.Log;
 
-import java.nio.ByteBuffer;
-
 /**
  * Created by Katarina Sheremet on 6/8/16 1:04 PM.
```
Comment (dotdoom) [inline app/src/main/java/org/dasfoo/rover/android/client/video/MediaCodecHandler.java]: Please describe the purpose of this class.

File: app/src/main/java/org/dasfoo/rover/android/client/video/StreamParser.java [hunk 2]
```diff
@@ -11,6 +11,12 @@
  * Take NAL units one by one from the specified InputStream.
  */
 public class StreamParser {
+
+    /**
+     * Maximum Nal unit size.
+     */
+    public static final int MAX_NAL_UNIT_SIZE = 65535;
```
Comment (dotdoom) [inline app/src/main/java/org/dasfoo/rover/android/client/video/StreamParser.java]: This is no longer necessary.

File: app/src/main/java/org/dasfoo/rover/android/client/video/VideoFragment.java [hunk 3]
```diff
@@ -69,72 +82,111 @@ public final View onCreateView(final LayoutInflater inflater, final ViewGroup co
...
+
+    /**
+     * Setters for idBufferQueue.
+     * Inserts the specified element at the tail of this queue,
+     * waiting if necessary for space to become available.
+     *
+     * @param id index of InputBuffer
+     * @throws InterruptedException if interrupted while waiting
+     */
+    public static void setIdBufferInQueue(final Integer id) throws InterruptedException {
```
Comment (dotdoom) [inline app/src/main/java/org/dasfoo/rover/android/client/video/VideoFragment.java]: `queueInputBufferId`

File: app/src/main/java/org/dasfoo/rover/android/client/video/VideoFragment.java [hunk 4]
```diff
@@ -69,72 +82,111 @@ public final View onCreateView(final LayoutInflater inflater, final ViewGroup co
...
+
+    /**
+     * Getters for idBufferQueue.
+     * Retrieves and removes the head of this queue,
+     * waiting if necessary until an element becomes available.
+     *
+     * @return head element of this queue
+     * @throws InterruptedException if interrupted while waiting
+     */
+    public static Integer getIdBufferFromQueue() throws InterruptedException {
```
Comment (dotdoom) [inline app/src/main/java/org/dasfoo/rover/android/client/video/VideoFragment.java]: `dequeueInputBufferId`

**Comment by dotdoom** [comment]
How about making `StreamParser` and `MediaCodec` private fields of `MediaCodecHandler`?  Then provide `MediaCodecHandler#getMediaCodec()` to call `#configure()` etc.

This will help later to split decoding into a separate library.

File: app/src/main/java/org/dasfoo/rover/android/client/video/StreamParser.java (hunk 2 above)
Comment (ksheremet) [inline app/src/main/java/org/dasfoo/rover/android/client/video/StreamParser.java]: Thanks! I haven't noticed that)

**Comment by ksheremet** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/55/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Comment by dotdoom** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/55/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Pull request merged**